pangaea/engine/data_preprocessor.py

```diff
@@ -1,324 +1,105 @@
-            self.dataset.data_min[modality] = new_stats[2]
-            self.dataset.data_max[modality] = new_stats[3]
-
-    def __getitem__(
-        self, index: int
+    def __call__(
+        self, data: dict[str, torch.Tensor | dict[str, torch.Tensor]]
     ) -> dict[str, torch.Tensor | dict[str, torch.Tensor]]:
-        """Return a modified item from the dataset.
-
-        Args:
-            index (int): index of data.
-        Returns:
-            dict[str, torch.Tensor | dict[str, torch.Tensor]]: output dictionary following the format
-            {"image":
-                {
-                encoder_modality_1: torch.Tensor of shape (C T H W) (with T=1 if single timeframe)
-                ...
-                encoder_modality_N: torch.Tensor of shape (C T H W) (with T=1 if single timeframe)
-                 },
-            "target": torch.Tensor of shape (H W),
-             "metadata": dict}.
-        """
-        data = self.dataset[index]
 
-        for k, v in data["image"].items():
-            if k in self.encoder.input_bands:
-                data["image"][k] = self.preprocessor[k](v)
-
-        data["target"] = data["target"].long()
-        return data
+        raise NotImplementedError
 
+    def update_meta(self, meta):
+        raise NotImplementedError
 
-class RegPreprocessor(SegPreprocessor):
-    def __init__(self, dataset: GeoFMDataset, encoder: Encoder) -> None:
-        """Initialize the RegPreprocessor for regression tasks."""
-        super().__init__(dataset, encoder)
-
-    def __getitem__(
-        self, index: int
-    ) -> dict[str, torch.Tensor | dict[str, torch.Tensor]]:
-        """Return a modified item from the dataset.
-
-        Args:
-            index (int): index of data.
-        Returns:
-            dict[str, torch.Tensor | dict[str, torch.Tensor]]: output dictionary following the format
-            {"image":
-                {
-                encoder_modality_1: torch.Tensor of shape (C T H W) (with T=1 if single timeframe)
-                ...
-                encoder_modality_N: torch.Tensor of shape (C T H W) (with T=1 if single timeframe)
-                 },
-            "target": torch.Tensor of shape (H W),
-             "metadata": dict}.
-        """
-        data = self.dataset[index]
+    def check_dimension(self, data: dict[str, torch.Tensor | dict[str, torch.Tensor]]):
+        """check dimension (C, T, H, W) of data"""
         for k, v in data["image"].items():
-            if k in self.encoder.input_bands:
-                data["image"][k] = self.preprocessor[k](v)
-        data["target"] = data["target"].float()
-        return data
+            if len(v.shape) != 4:
+                raise AssertionError(f"Image dimension must be 4 (C, T, H, W), Got {str(len(v.shape))}")
 
+        if len(data["target"].shape) != 2:
+           raise AssertionError(f"Target dimension must be 2 (H, W), Got {str(len(data['target'].shape))}")
```

Comment:
This requires single temporal datasets to return in the format (C, T, H, W), but not all of them have been updated accordingly.